inst/EpiModel 2.0 Introduction.rmd

```diff
@@ -0,0 +1,254 @@
+formation <- ~ edges + nodefactor("group")+nodematch("group")
+target.stats <- c(30, 30, 2)
+```
+
+In this network we have a total of 30 relationships in the population, two of which involve individuals of the same sex. We can replicate the traditional bipartite behavior of default network setups by setting the ```nodematch``` formation target statistic to 0:
+
+```{r echo = TRUE, eval = FALSE}
+target.stats <- c(30, 30, 0)
+```
+
+## Two-Group Functions
+
+Along with the change to how we build multi-group networks, we have also changed the built in EpiModel functions to make the workflow clearer (particularly for bespoke module development on the part of the end-user). In EpiModel 1.0, no distinction in module functionality was made for bipartite and non-bipartite networks. With EpiModel 2.0, we have split each of the appropriate core modules into one and two group sub-functions. For example, the base infection module is called as:
+
+```{r echo = TRUE, eval = FALSE}
+##One Group
+dat <- infection.net(dat, at)
+## Two Group
+dat <- infection.2g.net(dat, at)
+```
+
+based on the based on the input parameters passed through to ```init.net``` or ```param.net```. Input parameters for each of these two functions, for the second group, have also changed with EpiModel moving from mode to group language. Second group parameters or initial conditions are now denoted by a ```.g2``` suffix as opposed to the ```.m2``` suffix that was used previously:
+
+```{r echo = TRUE, eval = TRUE}
+init <- init.net(i.num = 10, i.num.g2 = 10)
+param <- param.net(inf.prob = 0.1, inf.prob.g2 = 0.2)
+```
+
+When epidemic type has been passed through to ```control.net``` through the ```type``` parameter, EpiModel will automatically select the appropriate version of base modules based on the presence of ```.g2``` parameters in either ```init``` or ```param``` objects.
+
+## Arrivals & Departures
+
+A further change made to the EpiModel workflow, was moving away from the language of "births" and "deaths" in models with vital dynamics, to "arrivals" and "departures" to account for a wider range of models in which entry into and exiting from the population is not neccessarily through a mortality process (e.g. entry and exit through an age process). To this end, all input parameters based on births and deaths have been renamed based on arrivals and departures. For example the birth rate ```b.rate``` passed to ```param.net()```, has been renamed ```a.rate``` for "arrival rate." The rate for deaths remains named ```d.rate``` though its interpretation has changed. Module functions based on birth and death rates have been changed: ```births.FUN``` and it's associated base module ```births.net``` are now ```arrivals.FUN``` and ```arrivals.net``` (```arrivals.2g.net``` in the two group case) respectively. Similar changes exists for ```deaths.FUN``` and ```deaths.net```, which are now ```departures.FUN``` and ```departures.net``` respectively.
```

Comment:
This is fine to mention, but it was actually implemented prior to EpiModel 2.0. So note that.